aiohttp/helpers.py

```diff
@@ -1038,3 +1039,19 @@ def parse_http_date(date_str: Optional[str]) -> Optional[datetime.datetime]:
             with suppress(ValueError):
                 return datetime.datetime(*timetuple[:6], tzinfo=datetime.timezone.utc)
     return None
+
+
+# Addresses can be either tuples of varying lengths (AF_INET, AF_INET6,
+# AF_NETLINK, AF_TIPC) or strings (AF_UNIX).
+def connect_socket_with_interface(
```

Comment:
This name seems misleading. Connecting a socket implies that it's pre-existing while I see that it's not true — it creates a socket and configures it in a specific way. Just like variable names, function names should tell a story about the effect the function has. And because it's a function, it represents an action and not a static object, keep using an imperative verb at the beginning.

How about something like this?
```suggestion
def make_an_interface_bound_socket(
```
Does it convey the purpose? Does it advertise why a caller would want to make use of it? If yes, it's a good name. If not, what could be done to improve it?

Reply:
Agreed, and co-authored you for the suggestion 👍